### PR TITLE
Filter the units by municipality if selected (TAM-142)

### PR DESCRIPTION
**Pull request by urshala**
If the municipality is selected, then filter out the units that
belongs to selected municipality in search view.

unit-municipality-filter

### CODE REVIEW

**Review by tuomas777 — APPROVED** [review]
LGTM 👍

File: src/domain/search/utils.js [hunk 1]
```diff
@@ -51,8 +51,11 @@ export const getApiParamsFromFilters = (filters) => {
   return params;
 };
 
-export const getUnitOptions = (units, locale) => {
-  const options = units.map((unit) => {
+export const getUnitOptions = (units, locale, municipality) => {
+  const unitsFilteredByMunicipality = (municipality
```
Comment (Niilis) [inline src/domain/search/utils.js]: I know this is react style JavaScript code, but I feel this takes an unnecessarily long amount of time to actually understand what is happening here.

I would refactor this in this way:
```js
let unitsFilteredByMunicipality = units;

if (municipality) {
    const municipalities = municipality.split(',');
    unitsFilteredByMunicipality = units.filter(unit => municipalities.includes(unit.municipality));
}

const options = unitsFilteredByMunicipality.map(unit => {
...
```

Each line above does one thing and understanding it should be considerably easier than fitting almost all functionality on one line. What do you think?

**Review by Niilis — CHANGES_REQUESTED** [review]
Changes are good, but I would refactor the utility function to be slightly more readable.

File: src/domain/search/utils.js (hunk 1 above)
Comment (urshala) [reply src/domain/search/utils.js]: I agree that it's more readable when spiltted to multiple lines. Made the suggested changes now

**Review by Niilis — APPROVED** [review]
Changes are good, branch is good to go! :shipit:

**Pull request merged**